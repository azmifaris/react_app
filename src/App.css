.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}
ul {
  padding-right: 40px;
  list-style-type: none;
  color:black;
}
a { color: inherit; } 

ba {
  width: 300px;
}

.App-header {
  background-color: white;
  min-height: 50vh;
  flex-direction: column;
  font-size: calc(10px + 2vmin);
  color:black;
}
.center{
  flex-direction: column;
  justify-content: center;
  align-items: center;
}
.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
h3 {
  padding-right: 30px;
}

z {
 padding-right: 30px;
}