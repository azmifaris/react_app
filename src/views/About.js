import React from 'react';

export default function About(){
    return(
        <div className="App">
            <header className="App-header">
                <h1>About</h1>

                <br/><h6>@azmifaris05@gmail.com</h6>
            </header>
        </div> 
    )
}