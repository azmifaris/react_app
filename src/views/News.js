import React, { useEffect } from 'react';

import {
    BrowserRouter as Router,
    Switch,
    Route,
    Link,
    useRouteMatch,
    useHistory
  } from 'react-router-dom';
  import Javascript from './javascript';
  import PHP from './PHP';
  import C from './c';

export default function News(){
    const history = useHistory();
    let match = useRouteMatch();
    useEffect(() => {
        window.scrollTo(0, window.innerHeight, 'smooth')
    }, [])
    console.log(match)
    return(
        <div className="App">
            <header className="App-header">
                    <h1>Materi</h1>
                    <ul>
                    <li><Link to={match.url+'/javascript'}>Java Script</Link></li>
                    <li><Link to={match.url+'/PHP'}>PHP</Link></li>
                    <li><Link to={match.url+'/C++'}>C++</Link></li>
                    </ul>
            </header>
            <Switch>
                    <Route  path ={match.url+'/javascript'}>
                        <center><Javascript/></center>
                    </Route>
                    <Route  path ={match.url+'/PHP'}>
                        <center><PHP /></center>
                    </Route>
                    <Route  path ={match.url+'/C++'}>
                        <center><C /></center>
                    </Route>
                </Switch>
        </div> 
    )
}