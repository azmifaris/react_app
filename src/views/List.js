import React,{Component} from 'react';
import '../App.css';
//class ProductTypeRow
class ProductTypeRow extends React.Component {
    render() {
      const type = this.props.type;
      return (
        <tr>
          <th colSpan="2">
            {type}
          </th>
        </tr>
      );
    }
  }
  //class ProductRow
  class ProductRow extends React.Component {
    render() {
      const product = this.props.product;
      const nama = product.ready ?
        product.nama :
        <span style={{color: 'red'}}>
          {product.nama}
        </span>;
  
      return (
        <tr>
          <td>{nama}</td>
          <td>{product.harga}</td>
        </tr>
      );
    }
  }
  //class ProductTable
  class ProductTable extends React.Component {
    render() {
      const filterText = this.props.filterText;
      const inStockOnly = this.props.inStockOnly;
  
      const rows = [];
      let lastType = null;
  
      this.props.products.forEach((product) => {
        if (product.nama.indexOf(filterText) === -1) {
          return;
        }
        if (inStockOnly && !product.ready) {
          return;
        }
        if (product.type !== lastType) {
          rows.push(
            <ProductTypeRow
              type={product.type}
              key={product.type} />
          );
        }
        rows.push(
          <ProductRow
            product={product}
            key={product.nama}
          />
        );
        lastType = product.type;
      });
  
      return (
        <table>
          <thead>
            <tr>
              <th>nama</th>
              <th>harga</th>
            </tr>
          </thead>
          <tbody>{rows}</tbody>
        </table>
      );
    }
  }
  //class SearchBar untuk proses pencarian
  class SearchBar extends React.Component {
    constructor(props) {
      super(props);
      this.handleFilterTextChange = this.handleFilterTextChange.bind(this);
      this.handleInStockChange = this.handleInStockChange.bind(this);
    }
    
    handleFilterTextChange(e) {
      this.props.onFilterTextChange(e.target.value);
    }
    
    handleInStockChange(e) {
      this.props.onInStockChange(e.target.checked);
    }
    
    render() {
      return (
        <form>
          <input
            type="text"
            placeholder="Search..."
            value={this.props.filterText}
            onChange={this.handleFilterTextChange}
          />
          <p>
            <input
              type="checkbox"
              checked={this.props.inStockOnly}
              onChange={this.handleInStockChange}
            />
            {' '}
            Only show products in stock
          </p>
        </form>
      );
    }
  }
  //class FilterableProductTable untuk mefilter produk yang tersedua
  class FilterableProductTable extends React.Component {
    constructor(props) {
      super(props);
      this.state = {
        filterText: '',
        inStockOnly: false
      };
      
      this.handleFilterTextChange = this.handleFilterTextChange.bind(this);
      this.handleInStockChange = this.handleInStockChange.bind(this);
    }
  
    handleFilterTextChange(filterText) {
      this.setState({
        filterText: filterText
      });
    }
    
    handleInStockChange(inStockOnly) {
      this.setState({
        inStockOnly: inStockOnly
      })
    }
  
    render() {
      return (
        <div>
          <SearchBar
            filterText={this.state.filterText}
            inStockOnly={this.state.inStockOnly}
            onFilterTextChange={this.handleFilterTextChange}
            onInStockChange={this.handleInStockChange}
          />
          <ProductTable
            products={this.props.products}
            filterText={this.state.filterText}
            inStockOnly={this.state.inStockOnly}
          />
        </div>
      );
    }
  }
  //kelas list 
class List extends Component{
    render(){
      //deskripsi product
        const PRODUCTS = [
            {type: "Mobil", harga: "Rp 148.000.000", ready: true, nama: "Brio"},
            {type: "Mobil", harga: "Rp 205.500.000", ready: true, nama: "Mobilio"},
            {type: "Mobil", harga: "Rp 252.500.000", ready: false, nama: "Jazz"},
            {type: "Motor", harga: "Rp 16.000.000", ready: true, nama: "Beat"},
            {type: "Motor", harga: "Rp 29.000.000", ready: false, nama: "PCX"},
            {type: "Motor", harga: "Rp 35.000.000", ready: true, nama: "CBR150R"}
          ];
          
          
      return (
        <div className="App">
            <h1>List Produk</h1>
            
            <center><FilterableProductTable products={PRODUCTS} /></center>
        </div>
      );
  
    };
    
  }
  export default List;


