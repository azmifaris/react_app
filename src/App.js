import React from 'react';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link
} from 'react-router-dom';
import { Home, About, Users  } from './views' ;
import News from './views/News';
import List from './views/List';
import './App.css';


function App() {
  return (
    <Router>
     <div className="App">
      <header className="App-header">
          <z><Link to="/">Home</Link></z>
          <z><Link to="/news">Materi</Link></z>
          <z> <Link to="/topics">Aplikasi</Link></z>
          <z><Link to="/list">List Produk</Link></z>
          <z> <Link to="/about">About</Link></z>
      {/* route */}
      <Switch>
        <Route exact path = "/">
          <Home />
        </Route>
        <Route path = "/about">
          <About />
        </Route>
        <Route path = "/topics">
          <Users />
        </Route>
        <Route path = "/news">
          <News></News>
        </Route>
        <Route path = "/list">
          <List />
        </Route>
        
      </Switch>
      </header>
      
      
      
 
    
    
    </div> 
    
       </Router>  
    
  );
}

export default App;
