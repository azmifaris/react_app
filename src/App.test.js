import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';
import Topics from './views/Users';
import News from './views/News';
import List from './views/List';
import About from './views/About';

// Jest-DOM --> DOM yang ada di Browser

test('renders learn react link', () => {
  const { getByText } = render(<App />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders learn react link', () => {
  const { getByText } = render(<About />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
test('renders learn react user', () => {
  const { getByText } = render(<Topics />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('renders learn react user', () => {
  const { getByText } = render(<News />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});
test('renders learn react user', () => {
  const { getByText } = render(<List />);
  const linkElement = getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('change lenguage', () => {
  const component = renderer.create(
    <App />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance()._handleClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  component.getInstance()._handleClick();
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});


test('object equality', () => {
  const iniObject = { pertama: 1 };
  iniObject['kedua'] = 2;
  expect(iniObject).toEqual({
    pertama: 1,
    kedua: 2
  });

  let sesuatu;
  expect(sesuatu).toBeUndefined();

  let iniNull = null;
  expect(iniNull).toBeNull();
  expect(iniNull).toBeDefined();
  expect(false).toBeFalsy();
  expect(true).toBeTruthy();

  let number = 4;
  expect(number).toBeGreaterThan(3);
  expect(number).toBeGreaterThanOrEqual(3);
  expect(number).toBeLessThan(5);


  let string = 'ini string';
  expect(string).not.toMatch(/a/);
  expect(string).toMatch(/i/);

  let array = [1, 2, 3];
  expect(array).toContain(1);
});