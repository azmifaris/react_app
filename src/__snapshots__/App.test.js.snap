// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change lenguage 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <z>
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </z>
    <z>
      <a
        href="/news"
        onClick={[Function]}
      >
        Materi
      </a>
    </z>
    <z>
      <a
        href="/topics"
        onClick={[Function]}
      >
        Aplikasi
      </a>
    </z>
    <z>
      <a
        href="/list"
        onClick={[Function]}
      >
        List Produk
      </a>
    </z>
    <z>
      <a
        href="/about"
        onClick={[Function]}
      >
        About
      </a>
    </z>
    <div>
      <h1>
        Selamat Datang
      </h1>
      <h5>
        silahkan masukan data anda
      </h5>
      <form
        onSubmit={[Function]}
      >
        <label>
          Nama :
          <input
            name="name"
            onChange={[Function]}
            type="text"
            value=""
          />
          <br />
          Skill
          <select
            multiple={true}
            onChange={[Function]}
            value={Array []}
          >
            <option
              value="memasak"
            >
              Memasak
            </option>
            <option
              value="menari"
            >
              Menari
            </option>
            <option
              value="menyanyi"
            >
              Menyanyi
            </option>
          </select>
          <br />
          Alamat
          <textarea
            name="alamat"
            onChange={[Function]}
            value=""
          />
          <br />
          <button
            type="submit"
          >
            Submit
          </button>
        </label>
      </form>
    </div>
  </header>
</div>
`;
